modules/standard/GPU.chpl

```diff
@@ -336,50 +336,50 @@ module GPU
     checkValidGpuAtomicOp(opName, rtName, T);
 
     pragma "codegen for GPU"
-    extern rtName proc chpl_atomicBinOp(x, val);
+    extern rtName proc chpl_atomicBinOp(x, val) : T;
 
     __primitive("chpl_assert_on_gpu", false);
-    chpl_atomicBinOp(c_ptrTo(x), val);
+    return chpl_atomicBinOp(c_ptrTo(x), val);
   }
 
   private inline proc gpuAtomicTernOp(param opName : string, ref x : ?T, cmp : T, val : T) {
     param rtName = externFunc(opName, T);
     checkValidGpuAtomicOp(opName, rtName, T);
 
     pragma "codegen for GPU"
-    extern rtName proc chpl_atomicTernOp(x, cmp, val);
+    extern rtName proc chpl_atomicTernOp(x, cmp, val) : T;
 
     __primitive("chpl_assert_on_gpu", false);
-    chpl_atomicTernOp(c_ptrTo(x), cmp, val);
+    return chpl_atomicTernOp(c_ptrTo(x), cmp, val);
   }
 
   /* When run on a GPU, atomically add 'val' to 'x' (result is stored in 'x'). */
```

Comment:
Since these comments are used for the chpldoc: could you update them all to indicate that we return the old value.

Reply:
I updated the documentation regarding the old values.

I also added a couple of things regarding the behavior of `gpuAtomicInc` and `gpuAtomicDec` based on the [Nvidia documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html?highlight=atomics#atomicinc). I also removed the `nodoc` tag on `gpuAtomicExch` since to my knowledge there wasn't a good reason to only have that one being nodoc'ed. Please correct me if I'm wrong.

Reply:
> I also removed the nodoc tag on gpuAtomicExch since to my knowledge there wasn't a good reason to only have that one being nodoc'ed. Please correct me if I'm wrong

See here: https://github.com/chapel-lang/chapel/pull/22556

"I now have doc comments for the various atomic functions. With the notable exception of gpuAtomicExch, which I don't think is one is too terribly useful unless it also returns the old value (which we don't do at the moment)."

I suppose you've now updated to return the old value so doc'ing it looks like the right thing to do. Thanks!